src/app/public/src/modules/clipboard/clipboard.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input } from '@angular/core';
+import { StacheCopyToClipboardService } from './clipboard.service';
+import { timer } from 'rxjs/observable/timer';
+
+@Component({
+  selector: 'stache-copy-to-clipboard',
+  templateUrl: './clipboard.component.html'
+})
+export class StacheCopyToClipboardComponent {
+  @Input()
+  public copyTarget: HTMLElement;
+
+  @Input()
+  public buttonText: string;
+
+  public buttonActive: boolean = false;
+
+  constructor(
+    private clipboardService: StacheCopyToClipboardService
```

Comment:
@Blackbaud-TimPepper @Blackbaud-BrandonJones I concur with Bobby's request to change the button's text to "Copied!" after a successful operation.

This can be done easily with the `SkyAppResourcesService`.

**clipboard.component.ts**

```
this.resourcesService.getString('stache_clipboard_copied_button').subscribe((value) => {
  this.buttonText = value;
});
```

Reply:
This would also require us to support a second user input for `copiedText` from the user.  Also the [microsoft docs](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/?view=aspnetcore-2.1&tabs=visual-studio) don't change the value either.  Not sure it's necessary?